src/main/webapp/dev/js/common/questionRubric.js

```diff
@@ -387,10 +378,10 @@ function moveAssignWeightsCheckbox(checkbox) {
  * appropriate location
  */
 function bindAssignWeightsCheckboxes() {
-    $('body').on('click', 'input[id^="rubricAssignWeights"]', function () {
+    $('body').on('change', 'input[id^="rubricAssignWeights"]', function () {
         const $checkbox = $(this);
 
-        $checkbox.closest('form').find('tr[id^="rubricWeights"]').toggle();
+        $checkbox.closest('form').find('div > table > thead > tr[id^="rubricWeights"]').toggle();
```

Comment:
Can move the logic to `toggleAssignWeightsRow(checkbox)`

Reply:
Anything wrong with the previous `find('tr[id^="rubricWeights"]')`? We should not assume the DOM structure (Recall `>` CSS selector: Selects all elements where the parent is a element).

Reply:
>Anything wrong with the previous `find('tr[id^="rubricWeights"]')`?

The `.toggle()` function was not working using the previous selector after I moved the checkbox out of the grid. Which is why I used this. Let me have a look at it again.

Reply:
>Can move the logic to `toggleAssignWeightsRow(checkbox)`?

Moving the `.toggle()` to `toggleAssignWeightsRow(checkbox)` causes problems as the function is also called in `disableQuestion()` and `enableNewQuestion()` functions directly, which also toggles the weight row where it shouldn't.

Reply:
Move to the `if` statement. Change the `.toggle()` to `show/hide`?